### PR TITLE
[Intel XPU] Initially add nightly GSM8K accuracy tests for Llama-3.1-8B (TP=2) and Qwen3-32B (TP=4)

### PR DESCRIPTION
**Pull request by arathi-hlab**
Wire up the Intel XPU nightly flow on `intel-bmg-nightly`. Every suite/job is registered via `register_xpu_ci(... nightly=True)` and dispatched by `test/run_suite.py`, where the digit in the suite name = TP size, not host card count.

## Suites and tests

| Suite | TP | Test | Threshold |
|---|---|---|---|
| `nightly-xpu-2-gpu` | 2 | `test_xpu_llama_3_1_8b.py` | GSM8K ≥ 0.80 |
| `nightly-xpu-4-gpu` | 4 | `test_xpu_qwen3_32b.py` | GSM8K ≥ 0.85 |

Both jobs serialize on the single `intel-bmg-nightly` host. A `check-all-jobs` aggregator fails the workflow if any nightly suite job failed or was cancelled.

## Plumbing

- `python/sglang/test/xpu/simple_eval_gsm8k_xpu_mixin.py` launches the SGLang server with `--device xpu --attention-backend intel_xpu`, runs `simple_eval_gsm8k.GSM8KEval` (same evaluator as the AMD/CUDA nightlies), asserts `accuracy >= threshold`, and renders a Markdown table to `$GITHUB_STEP_SUMMARY`.
- `python/sglang/test/xpu/test_xpu_utils.py` writes the per-run summary table.
- New input `continue_on_error` (default `true`) on `workflow_dispatch` / `workflow_call` so a single flaky model doesn't fail the whole pipeline; the schedule trigger always passes `--continue-on-error` to `run_suite.py`.

## Test plan

- [ ] `workflow_dispatch` against this branch — verify both jobs spin up on `intel-bmg-nightly` and `check-all-jobs` reflects each result.
- [ ] Confirm `python3 run_suite.py --hw xpu --suite nightly-xpu-{2,4}-gpu --nightly` discovers exactly the registered tests for each suite.
- [ ] Spot-check the rendered `$GITHUB_STEP_SUMMARY` table after one full run; tune accuracy thresholds in a follow-up if any model is consistently below baseline.

---
### CI States

Latest PR Test (Base): :hourglass_flowing_sand: [Run #28496829842](https://github.com/sgl-project/sglang/actions/runs/28496829842)
Latest PR Test (Extra): :hourglass_flowing_sand: [Run #28496829739](https://github.com/sgl-project/sglang/actions/runs/28496829739)

### CODE REVIEW

File: test/registered/xpu/llm_models/test_xpu_qwen3_32b.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import unittest
+
+import torch
+
+from sglang.test.ci.ci_register import register_xpu_ci
+from sglang.test.test_utils import CustomTestCase
+from sglang.test.xpu.simple_eval_gsm8k_xpu_mixin import SimpleEvalGSM8KXPUMixin
+
+register_xpu_ci(est_time=1800, suite="nightly-xpu-4-gpu", nightly=True)
```
Comment (mingfeima) [inline test/registered/xpu/llm_models/test_xpu_qwen3_32b.py:15]: wow, this one is huge, does our CI machines can take it? how long each CI task take now?

File: .github/workflows/nightly-test-intel.yml [hunk 2]
```diff
@@ -2,32 +2,180 @@ name: Nightly Test (Intel)
...
       - main
     paths:
       - "python/sglang/version.py"
   workflow_dispatch:
+    inputs:
+      continue_on_error:
+        description: 'Continue on error (do not fail the workflow on test failures)'
+        required: false
+        type: boolean
+        default: true
```
Comment (mingfeima) [inline .github/workflows/nightly-test-intel.yml:18]: could you please explain the intention of `continue-on-error` behavior? This will let test failures pass without turning the workflow red?

**Review by mingfeima — COMMENTED** [review]
* continue-on-error why?
* ci test time change?

**Comment by mingfeima** [comment]
cc @MingxuZh

File: test/registered/xpu/llm_models/test_xpu_qwen3_32b.py (hunk 1 above)
Comment (arathi-hlab) [reply test/registered/xpu/llm_models/test_xpu_qwen3_32b.py:15]: this is for nightly flow not for ci so on nightly continue -on -error is accepted

File: .github/workflows/nightly-test-intel.yml (hunk 2 above)
Comment (arathi-hlab) [reply .github/workflows/nightly-test-intel.yml:18]: Since this is the nightly flow, I want to run all UTs/models regardless of individual test failures. If one test fails, the pipeline should continue executing the remaining tests instead of stopping. Otherwise, the remaining tests won't run, and we won't be able to identify regressions across the full test suite as this is nightly flow not the ci flow for that we have dedicated runner (for nightly flow) as we are passing (nightly= True)

File: test/registered/xpu/llm_models/test_xpu_qwen3_32b.py (hunk 1 above)
Comment (arathi-hlab) [reply test/registered/xpu/llm_models/test_xpu_qwen3_32b.py:15]: in nightly we have suites and inside suites we can run multiple models and each model we can set it out  time

**Review by mingfeima — APPROVED** [review]
